### ISSUE TITLE
HighResolutionIconUrl invalid on creation

### ISSUE THREAD

**Issue by jthake** [OfficeDev/generator-office#48]
Getting an error in uploading the manifest.xml when using yo office.

"This app can't be installed. The HighResolutionIconUrl value '[Icon URL]' isn't a corectly formatted URL."

 Although it's great we use these values. I think its better to actually include a icon placeholder in the project and reference that with https://localhost:8443/ url as we are hard coding that anyway. Less steps to getting it running that way.

**Comment by MrPiao** [comment]
I have a local fix using an icon in the public domain but having problems with verification...

**Comment by jasonjoh** [comment]
I ran into this as well. Fixing `HighResolutionIconUrl` will cause it to then fail on `SupportUrl`. So you actually need to fix both values before it will work out of the box.

**Comment by MrPiao** [comment]
Yeah my pull requests fixes both :)

**Comment by jasonjoh** [comment]
Oh ok. I only looked at the referenced commit.

**Comment by MrPiao** [comment]
Must have forgotten to ref it on the second commit.

**Comment by andrewconnell** [comment]
Hmmm.... this PR only fixed the mail addin & doesn't include any tests to verify the image file was correctly added to the target source by the generator. What about task pane & content pane addins?

As a general practice, IMHO fixes that should be global to all addins (such as this one) should not be added to **master** & published to NPM until the change is implemented in all relevant addins & it doesn't cause a drop of test code coverage.

**Comment by jthake** [comment]
@MrPiao are you ok to update the task and content pane ones and add the test coverage?

**Comment by MrPiao** [comment]
Ah, I didn't even think about the other add-in types. Call it a side effect of working on the Outlook team :D

I should be able to do this for task pane/content add-ins. I can also add the tests - really should've done in the first place :)

**Comment by andrewconnell** [comment]
@MrPiao So the fact the PR with this one in #53 causes an issue because it blocks us from merging in other open PR's. We don't want to include the code in #53 until it's really complete...

So when I merge the other PR's in, I'm going to back out the changes that were merged in in PR #53 and ask you resubmit them when you have the task pane, content addin & tests to make this more complete.

Cool?

**Comment by MrPiao** [comment]
Cool


Alex Park
alexpark.info<http://alexpark.info>
@AayPizzle
________________________________

**Comment by jasonjoh** [comment]
Guys, can we reopen this one? I'll submit a fix for it.

**Comment by jthake** [comment]
Reopened. will get time late r on this. I aggree this drives me nuts ;-)

**Comment by jasonjoh** [comment]
Ok, I've worked up some test cases. I was reading the thread and one of the sticking points with @MrPiao's PR was he didn't do anything with Task and Content addins. I looked at the template manifests and those addin types don't set `HighResolutionIconUrl` at all, so they wouldn't fail when uploading. So I'm not writing tests for those types, that seems out of the scope of this particular issue.

According to https://msdn.microsoft.com/EN-US/library/office/fp161044.aspx `HighResolutionIconUrl` isn't required for Task and Content addins, so that seems like a non-issue.
